ql/src/test/queries/clientpositive/compaction_query_based_insert_only_clustered.q

```diff
@@ -0,0 +1,33 @@
+--! qt:replace:/createTime:(\d+)/#Masked#/
+--! qt:replace:/location:(\S+)/#Masked#/
+--! qt:replace:/lastAccessTime:(\d+)/#Masked#/
+--! qt:replace:/ownerType:(\S*)/#Masked#/
+--! qt:replace:/owner:(\S*)/#Masked#/
+--! qt:replace:/skewedColValueLocationMaps:(\S*)/#Masked#/
+--! qt:replace:/transient_lastDdlTime=(\d+)/#Masked#/
+--! qt:replace:/totalSize=(\d+)/#Masked#/
+--! qt:replace:/rawDataSize=(\d+)/#Masked#/
+--! qt:replace:/writeId:(\d+)/#Masked#/
+--! qt:replace:/bucketing_version=(\d+)/#Masked#/
+--! qt:replace:/id:(\d+)/#Masked#/
+
+drop table orc_bucketed;
+
+create table orc_bucketed (a int, b string) clustered by (a) into 3 buckets stored as orc TBLPROPERTIES('transactional'='true', 'transactional_properties'='insert_only');
+
+insert into orc_bucketed values('1', 'text1');
+insert into orc_bucketed values('2', 'text2');
+insert into orc_bucketed values('3', 'text3');
+insert into orc_bucketed values('4', 'text4');
+insert into orc_bucketed values('5', 'text5');
+insert into orc_bucketed values('6', 'text6');
+insert into orc_bucketed values('7', 'text7');
+insert into orc_bucketed values('8', 'text8');
+insert into orc_bucketed values('9', 'text9');
+insert into orc_bucketed values('10', 'text10');
+
+describe extended orc_bucketed;
+alter table orc_bucketed compact 'MAJOR' and wait;
+analyze table orc_bucketed compute statistics;
```

Comment:
Is `analyze table compute statistics` command required here? 
Stats update usually happens within the compaction cycle so I think re-evaluation of stats is extra effort.

The place where Stats update happens in compaction is here - https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/StatsUpdater.java

Reply:
Thanks for the review @SourabhBadhya, The reason why this is required here is to make sure the test runs independent of Hive config. The automatic Stats updates happens when hive.stats.autogather=true.
To ensure we have the data in q.out regardless of config analyze statement is used. Let me know if you have any thought on this.

Reply:
Stats update on query based compaction depends upon this config - 
`hive.compactor.gather.stats`
Source - https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java#L103
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java#L444

After taking a deeper look, this config seems to be disabled in MiniLlapLocalCompactorCliDriver here as part of HIVE-26802 - 
https://github.com/apache/hive/blob/master/itests/util/src/main/java/org/apache/hadoop/hive/cli/control/CliConfigs.java#L271

@zratkai Is there any reason why this was disabled as part of HIVE-26802? I did try some tests locally and I was able to see correct stats after removing this disabling config line from CliConfigs.

Reply:
@SourabhBadhya IMK hive.compactor.gather.stats is used for Compaction stats.
the analyze table command is for table stats and for auto generation of those stats `hive.stats.autogather=true` is used.
Ref: https://docs.cloudera.com/HDPDocuments/HDP3/HDP-3.1.4/performance-tuning/content/hive_generate_hive_statistics.html.
Please correct me if Im wrong

Reply:
@akshat0395 `hive.stats.autogather` is already true by default. This config is used by insert/delete/update statements to compute stats once records are written. 

Compaction checks this config - `hive.compactor.gather.stats` to see whether it has to perform stats computation of the table which has been compacted. Since its turned off by MiniLlapLocalCompactorCliDriver, stats computation is disabled.

Reply:
I do not remember exactly why it was necessary to do it this way. With this steps you can force the analyze to happen exactly when it needed. If I remember correctly the autogather is async, and it caused issues, that sometimes didn't happen until the test needed at the last describe table, so it failed.

Reply:
@zratkai Ok makes sense then we can use the `analyze table compute statistics` if it causes delay issues.